### PR TITLE
changelog: document changes since 0.27

### PR DESCRIPTION
**Pull request by ethomson**

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -3,16 +3,79 @@ v0.27 + 1
...
+
+* When an HTTP server supports both NTLM and Negotiate authentication
+  mechanisms, we would previously fail to authenticate with any mechanism.
+
+* The `GIT_OPT_SET_PACK_MAX_OBJECTS` option can now set the maximum
+  number of objects allowed in a packfile being downloaded; this can help
+  limit the maximum memory used when fetching from an untrusted remote.
+
+* Line numbers in diffs loaded from patch files were not being populated;
+  they are now included in the results..
```
Comment (tiennou) [inline docs/changelog.md]:
```suggestion
  they are now included in the results.
```

File: docs/changelog.md [hunk 2]
```diff
@@ -3,16 +3,79 @@ v0.27 + 1
...
+  `GIT_BLAME_USE_MAILMAP` option.
+
+* Ignore handling for files in ignored folders would be ignored.
+
+* Worktrees can now be backed by bare repositories.
+
+* Trailing spaces are supported in `.gitignore` files, these spaces were
+  previously (and erroneously) treated as part of the pattern.
+
+* The library can now be built with mbedTLS support for HTTPS.
```
Comment (tiennou) [inline docs/changelog.md]: I think that was in 0.26 ?

File: docs/changelog.md [hunk 3]
```diff
@@ -28,10 +91,34 @@ v0.27 + 1
...
 
+* Reference specifications can be parsed from an arbitrary string with
+  the `git_refspec_parse` API.
+
+* You can now get the name and path of worktrees using the
+  `git_worktree_name` and `git_worktree_path` APIs, respectively.
+
+* The `ref` field has been added to `git_worktree_add_options` to enable
+  the creation of a worktree from a pre-existing branch.
+
```
Comment (tiennou) [inline docs/changelog.md]:
```suggestion

* It's now possible to analyze merges between any references, not
  just against `HEAD`, using `git_merge_analysis_for_ref`.
```

File: docs/changelog.md (hunk 2 above)
Comment (ethomson) [reply docs/changelog.md]: I don't think so?  We didn't actually end up merging it until early 2018 and I don't see mbedTLS in the `maint/v0.27` branch...?

File: docs/changelog.md (hunk 3 above)
Comment (ethomson) [reply docs/changelog.md]: Thanks, added.

File: docs/changelog.md (hunk 1 above)
Comment (ethomson) [reply docs/changelog.md]: Thanks!

**Pull request merged**